x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/User.java

```diff
@@ -9,23 +9,12 @@
 import org.elasticsearch.xpack.sql.session.Configuration;
 import org.elasticsearch.xpack.sql.tree.Location;
 import org.elasticsearch.xpack.sql.tree.NodeInfo;
+import org.elasticsearch.xpack.sql.type.DataType;
 
-import java.util.Objects;
-
-public class User extends BaseSystemFunction {
+public class User extends ConfigurationFunction {
 
     public User(Location location, Configuration configuration) {
-        super(location, configuration);
-    }
-
-    @Override
-    protected NodeInfo<User> info() {
-        return NodeInfo.create(this, User::new, configuration());
-    }
-
-    @Override
-    public boolean nullable() {
-        return true;
+        super(location, configuration, DataType.KEYWORD);
```

Comment:
USER() function can return `null` when Security is disabled...

Reply:
Then it should not be a constant.